core/src/test/java/io/questdb/griffin/AlterTableWalEnabledTest.java

```diff
@@ -0,0 +1,175 @@
+        });
+    }
+
+    @Test
+    public void testWalEnabledAddIndex() throws Exception {
+        assertMemoryLeak(() -> {
+            String alterSuffix = "ALTER COLUMN s ADD INDEX";
+            checkWalEnabledBeforeAfterAlter(alterSuffix);
+        });
+    }
+
+    @Test
+    public void testWalEnabledAndAlterLag() throws Exception {
+        assertMemoryLeak(() -> {
+            String alterSuffix = "set param commitLag=100s";
+            checkWalEnabledBeforeAfterAlter(alterSuffix);
+        });
+    }
+
+    @Test
+    public void testWalEnabledAndRenameColumn() throws Exception {
+        assertMemoryLeak(() -> {
+            String alterSuffix = "rename column x to y";
+            checkWalEnabledBeforeAfterAlter(alterSuffix);
+        });
+    }
+
+    @Test
+    public void testWalEnabledNameInCreateAsSelect() throws Exception {
+        assertMemoryLeak(() -> {
+            compile("create table wm as (" +
+                    "select x, cast(x as timestamp) as ts " +
+                    "from long_sequence(2) " +
+                    ") timestamp(ts) partition by DAY WAL");
+
+            assertWalEnabled("wm", true);
+        });
+    }
+
+    @Test
+    public void testWalEnabledNameInCreateAsSelect2() throws Exception {
+        assertMemoryLeak(() -> {
+            compile("create table wm as (" +
+                    "select x, cast(x as timestamp) as ts " +
+                    "from long_sequence(2) " +
+                    ") timestamp(ts) partition by DAY Bypass WaL");
+
+            assertWalEnabled("wm", false);
+        });
+    }
+
+    @Test
+    public void testWalEnabledNameInvalid() throws Exception {
+        assertMemoryLeak(() -> {
+            try {
+                createTableWrite("my_table_wal", "NONE", "DAY");
+                Assert.fail("Exception expected");
+            } catch (SqlException ex) {
+                TestUtils.assertContains(
+                        ex.getFlyweightMessage(),
+                        "unexpected token: NONE"
+                );
+            }
+        });
+    }
+
+    @Test
+    public void testWalEnabledNameInvalidWalWord() throws Exception {
+        assertMemoryLeak(() -> {
+            try {
+                createTableWrite("my_table_wal", "BYPASS WALL", "DAY");
+                Assert.fail("Exception expected");
+            } catch (SqlException ex) {
+                TestUtils.assertContains(
+                        ex.getFlyweightMessage(),
+                        "invalid syntax, should be BYPASS WAL but was BYPASS WALL"
+                );
+            }
+        });
+    }
+
+    @Test
+    public void testWalEnabledNonPartitionedTable() throws Exception {
+        assertMemoryLeak(() -> {
+            try {
+                createTableWrite("my_table_wal", "WAL", "NONE");
+                Assert.fail("Exception expected");
+            } catch (SqlException ex) {
+                TestUtils.assertContains(
```

Comment:
error position should be checked and asserted

Reply:
fixed in fixed in https://github.com/questdb/questdb/pull/2335/commits/bf37ae5577a61e21bd8604025c24c6259b0cca1a